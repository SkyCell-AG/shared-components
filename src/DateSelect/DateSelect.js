import React, {
    useCallback,
} from 'react'
import PropTypes from 'prop-types'
import {
    FormHelperText,
    TextField,
} from '@mui/material'
import {
    DatePicker,
} from '@mui/lab'
import {
    useTheme,
} from '@mui/material/styles'
import clsx from 'clsx'

import Label from 'Label'

import {
    strToDate,
} from 'utils/DateUtils'

import useStyles from './DateSelect.styles'

const propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    hasError: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    required: PropTypes.bool,
    value: PropTypes.string,
    className: PropTypes.string,
}

const defaultProps = {
    title: '',
    name: '',
    hasError: '',
    disabled: false,
    required: false,
    value: null,
    className: '',
}

const DateSelect = (props) => {
    const theme = useTheme()
    const classes = useStyles(theme)
    const {
        title,
        name,
        hasError,
        onChange,
        value,
        disabled,
        required,
        className,
    } = props

    const handleOnChange = useCallback((date) => {
        const newValue = new Date(date)

        onChange(newValue, {
            target: {
                name,
                value: name === 'from' ? new Date(newValue.setHours(0, 0, 0)) : new Date(newValue.setHours(23, 59, 59)),
            },
        })
    }, [
        name,
        onChange,
    ])

    return (
        <div className={className}>
            <Label
                title={title}
                hasError={hasError}
            />
            <div
                className={clsx(
                    classes.dateField,
                    {
                        [classes.disabled]: disabled,
                        [classes.isRequired]: required,
                    },
                )}
            >
                <div className={classes.requiredMark} />
                <DatePicker
                    className={classes.input}
                    value={strToDate(value)}
                    format="DD.MM.YYYY"
                    onChange={handleOnChange}
                    animateYearScrolling
                    error={Boolean(hasError)}
                    renderInput={(params) => {
                        return (
                            <TextField
                                {...params}
                                variant="standard"
                            />
                        )
                    }}
                    disabled={disabled}
                    fullWidth
                    data-testid={`dateSelect-${name}`}
                />
                <FormHelperText
                    error={Boolean(hasError)}
                    className={classes.errorMessage}
                >
                    {hasError}
                </FormHelperText>
            </div>
        </div>
    )
}

DateSelect.propTypes = propTypes
DateSelect.defaultProps = defaultProps

export default DateSelect
