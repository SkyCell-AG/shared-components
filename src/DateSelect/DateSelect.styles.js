import {
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles((theme) => {
    return {
        dateField: {
            position: 'relative',
        },
        input: {
            '& input': {
                cursor: 'pointer',
            },
        },
        requiredMark: {
            position: 'absolute',
            left: -8,
            width: 6,
            height: 26,
            backgroundColor: theme.palette.primary.main,
            display: 'none',
            borderRadius: 3,
        },
        isRequired: {
            '& $requiredMark': {
                display: 'block',
            },
            '& .MuiInputBase-input': {
                backgroundColor: theme.palette.primary[30],
            },
        },
        disabled: {
            '& $requiredMark': {
                backgroundColor: 'transparent !important',
            },
            '& .MuiInputBase-input': {
                backgroundColor: 'transparent !important',
            },
            '& .MuiInput-underline:before': {
                border: '0 !important',
            },
        },
        errorMessage: {
            height: 10,
        },
    }
})

export default useStyles
