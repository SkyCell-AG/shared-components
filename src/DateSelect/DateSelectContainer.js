import React from 'react'
import PropTypes from 'prop-types'

import DateSelect from './DateSelect'

const propTypes = {
    onChange: PropTypes.func.isRequired,
}

const DateSelectContainer = (props) => {
    return (
        <DateSelect
            {...props}
        />
    )
}

DateSelectContainer.propTypes = propTypes

export default DateSelectContainer
