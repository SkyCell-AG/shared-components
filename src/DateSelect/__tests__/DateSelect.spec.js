import React from 'react'
import {
    shallow,
} from 'enzyme'

import DateSelect from '../DateSelect'

describe('shared-components/DateSelect', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(
            <DateSelect
                value="26.06.2020"
                onChange={jest.fn()}
                name="validFrom"
            />,
        )

        expect(wrapper).toMatchSnapshot()
    })
})
