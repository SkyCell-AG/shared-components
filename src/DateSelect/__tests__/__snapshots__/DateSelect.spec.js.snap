// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared-components/DateSelect should match snapshot 1`] = `
<div
  className=""
>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <Label
      className=""
      hasError=""
      title=""
    />
    <div
      className="makeStyles-dateField-1"
    >
      <div
        className="makeStyles-requiredMark-3 makeStyles-requiredMark-7"
      />
      <PickerWithState
        allowKeyboardControl={true}
        animateYearScrolling={true}
        className="makeStyles-input-2"
        data-testid="dateSelect-validFrom"
        disabled={false}
        error={false}
        format="DD.MM.YYYY"
        fullWidth={true}
        invalidDateMessage="Invalid Date Format"
        maxDate={2100-01-01T00:00:00.000Z}
        maxDateMessage="Date should not be after maximal date"
        minDate={1900-01-01T00:00:00.000Z}
        minDateMessage="Date should not be before minimal date"
        onChange={[Function]}
        openTo="date"
        value={2020-06-26T00:00:00.000Z}
        views={
          Array [
            "year",
            "date",
          ]
        }
      />
      <WithStyles(ForwardRef(FormHelperText))
        className="makeStyles-errorMessage-6"
        error={false}
      />
    </div>
  </MuiPickersUtilsProvider>
</div>
`;
